src/pages/academy/components/search_banner/_search-banner.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { useEffect, useState } from 'react'
+import styled, { css } from 'styled-components'
+import { combined_filter_type } from '../../common/_constants'
+import { MainWrapper, MobileWrapper, ParentWrapper, StyledLink } from './_search-styled'
+import SearchNav from './_search-nav'
+import SearchBar from './_search-bar'
+import TopicItemsAccordion from './_topic-items-accordion'
+import { Container, Flex } from 'components/containers'
+import { Header } from 'components/elements'
+import { useBrowserResize } from 'components/hooks/use-browser-resize'
+import { slugify, isBrowser } from 'common/utility'
+import device from 'themes/device'
+import Chevron from 'images/svg/custom/chevron-thick.svg'
+import { useCountryRule } from 'components/hooks/use-country-rule'
+
+type SearchBannerProps = {
+    hidden?: boolean
+}
+
+const fadeKeyframes = css`
+    animation: fade 0.4s ease-in-out;
+    @keyframes fade {
+        0% {
+            opacity: 0;
+        }
+        100% {
+            opacity: 1;
+        }
+    }
+`
+const TopicParent = styled(Flex)`
+    display: ${(props) => (props.modal ? 'flex' : 'none')};
+    z-index: ${(props) => (props.modal ? '100' : 'unset')};
+    position: relative;
+    ${(props) => props.modal && fadeKeyframes}
+`
+const TopicWrapper = styled(Flex)`
+    position: absolute;
+    top: 0;
+    left: 0;
+    flex-wrap: wrap;
+    height: auto;
+    align-items: flex-start;
+    z-index: 100;
+    background: var(--color-white);
+    padding: 32px 0;
+
+    @media ${device.tabletL} {
+        padding: 24px 0;
+    }
+`
+const TopicItemsParentWrapper = styled(Container)`
+    flex-wrap: wrap;
+`
+const TopicItemWrapper = styled(Flex)`
+    max-width: 250px;
+    padding: 16px;
+    height: auto;
+
+    @media ${device.laptopM} {
+        max-width: 50%;
+    }
+
+    @media ${device.tabletL} {
+        display: ${(props) => (props.is_mobile_expanded ? 'none' : 'flex')};
+    }
+`
+
+const TopicMobileParentWrapper = styled(Flex)`
+    display: ${(props) => (props.is_mobile_expanded ? 'none' : 'flex')};
+`
+
+const SearchBanner = ({ hidden }: SearchBannerProps) => {
+    const [is_mobile] = useBrowserResize(768)
+    const [modal_opened, setModal] = useState(false)
+    const [hide_mobile_topic, setHideMobileTopic] = useState(false)
+    const [blog_post_url, setBlogPostURL] = useState(false)
+    const { is_eu, is_uk } = useCountryRule()
+
+    // Filter out restricted categories from the combined filter type array based on geolocation
+    useEffect(() => {
+        combined_filter_type.forEach((type) => {
```

Comment:
Same for here 
`
type.items = type.items.filter(({ is_visible_eu, is_visible_uk }) => {
                if (is_eu) return is_visible_eu
                if (is_uk) return is_visible_uk
                return obj
            })
`

Reply:
If I do it I cannot use `return obj` in last line.

Reply:
Alrighty